### ISSUE TITLE
BUG: Can not add new item and edit for Minimum Needs Manager in QGIS Bahasa Indonesia

### ISSUE THREAD

**Issue by harrymahar** [inasafe/inasafe#2676]
# Problem
Add new item using InaSAFE minimum needs manager in QGIS using Bahasa Indonesia does not work.

# Detail
In InaSAFE Training last week in The National Development University, Yogyakarta, I taught participant to **adjust Minimum needs** for Flood Impact Function using **Minimum needs manager** tool. In this training we using QGIS in Bahasa Indonesia because participants think that would be easier for them to understand all QGIS and InaSAFE tools in Bahasa Indonesia. Unfortunately, we had Python Error when we want to save new minimum need that we added into one of profiles in minimum needs manager. This problem occured not only for flood but other hazard impact function as well.

However, when I try to adjust minimum needs using QGIS in English Languange, It went well. I can save my new item and run it.

# Screenshoot
![minimum needs error](https://cloud.githubusercontent.com/assets/2534179/13733831/1b095e1e-e9ca-11e5-9276-8e13d9813146.PNG)
![minimum needs error2](https://cloud.githubusercontent.com/assets/2534179/13733833/1f387fd8-e9ca-11e5-992d-334e4fa29409.PNG)

# Platform
Windows 10 64 Bit
RAM 8 GB
QGIS 2.8.6 Wien (Long Term Release)
InaSAFE 3.2.6

cc : @timlinux @ismailsunni @samnawi

**Comment by samnawi** [comment]
yeah @harrymahar I also found this problem in my computer.
Also when I try to edit the english minimum need at QGIS with Bahasa language, the problem is occur.

`Terjadi kesalahan ketika menjalankan kode Python:


Traceback (most recent call last):
  File "/home/kurnia/.qgis2/python/plugins/inasafe/safe/gui/tools/minimum_needs/needs_manager_dialog.py", line 543, in save_resource
    parameter.name = resource['Resource name']
KeyError: 'Resource name'


Versi Python:
2.7.6 (default, Jun 22 2015, 18:01:27) 
[GCC 4.8.2]


Versi QGIS:
2.8.7-Wien Wien, exported

Jalur (path) Python: ['/usr/share/qgis/python/plugins/processing', '/usr/share/qgis/python', u'/home/kurnia/.qgis2/python', u'/home/kurnia/.qgis2/python/plugins', '/usr/share/qgis/python/plugins', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PILcompat', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.7', '/usr/lib/python2.7/dist-packages/ubuntu-sso-client', u'/home/kurnia/.qgis2//python', '/home/kurnia/.qgis2/python/plugins/OpenAerialMap', '/home/kurnia/.qgis2/python/plugins/OpenAerialMap/ext_libs/boto-2.38.0', '/home/kurnia/.qgis2/python/plugins/OpenAerialMap/ext_libs/filechunkio-1.6', '/home/kurnia/.qgis2/python/plugins/OpenAerialMap/ext_libs/requests-2.7.0', '/home/kurnia/.qgis2/python/plugins/inasafe', '/home/kurnia/.qgis2/python/plugins/inasafe/safe_extras/parameters', '/home/kurnia/.qgis2/python/plugins', '/home/kurnia/.qgis2/python/plugins/inasafe/safe_extras', '/usr/share/qgis/python/plugins/fTools/tools', '/home/kurnia/Documents/Kerja/InaSAFE Test/Sleman_Volcano']`

**Comment by harrymahar** [comment]
Hi, @timlinux @ismailsunni @samnawi @felix-yew

I currently testing new version of InaSAFE ( **InaSAFE 3.4.0 dev-master**) with HOT. I **Still** get this bug python error when I want to **Add New Item** and **Edit Item** in InaSAFE Minimum Needs Configuration.

This phtyon error when I want to **Add New Item**

```Traceback (most recent call last):
  File "C:/Users/Asus A450C/.qgis2/python/plugins\inasafe\safe\gui\tools\minimum_needs\needs_manager_dialog.py", line 543, in save_resource
    parameter.name = resource['Resource name']
KeyError: 'Resource name'


Versi Python:
2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]


Versi QGIS:
2.8.9-Wien Wien, 26c3ece

Jalur (path) Python: ['C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins\\processing', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python', u'C:/Users/Asus A450C/.qgis2/python', u'C:/Users/Asus A450C/.qgis2/python/plugins', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\matplotlib-1.3.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\nose-1.3.3-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\tornado-4.0.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\backports.ssl_match_hostname-3.4.0.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\certifi-14.05.14-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\bin\\python27.zip', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\DLLs', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\plat-win', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\lib-tk', 'C:\\PROGRA~1\\QGISWI~1\\bin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\PIL', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\jinja2-2.7.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\markupsafe-0.23-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\pytz-2012j-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Pythonwin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Shapely-1.2.18-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\wx-2.8-msw-unicode', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlrd-0.9.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlwt-0.7.5-py2.7.egg', u'C:/Users/Asus A450C/.qgis2//python', 'C:/Users/Asus A450C/.qgis2/python/plugins\\inasafe', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins\\inasafe\\safe_extras\\parameters', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins\\inasafe\\safe_extras', 'C:\\PROGRA~1\\QGISWI~1\\apps\\qgis-ltr\\python\\plugins\\fTools\\tools', 'D:/Work/OpenStreetMaps/InaSAFE/InaSAFE Testing 3.4/Testing/Tsunami']```

This Python Error when I want to **Edit Item**

```Terjadi kesalahan ketika menjalankan kode Python:


Traceback (most recent call last):
  File "C:/Users/Asus A450C/.qgis2/python/plugins\inasafe\safe\gui\tools\minimum_needs\needs_manager_dialog.py", line 543, in save_resource
    parameter.name = resource['Resource name']
KeyError: 'Resource name'


Versi Python:
2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]


Versi QGIS:
2.8.9-Wien Wien, 26c3ece

Jalur (path) Python: ['C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins\\processing', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python', u'C:/Users/Asus A450C/.qgis2/python', u'C:/Users/Asus A450C/.qgis2/python/plugins', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\matplotlib-1.3.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\nose-1.3.3-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\tornado-4.0.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\backports.ssl_match_hostname-3.4.0.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\certifi-14.05.14-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\bin\\python27.zip', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\DLLs', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\plat-win', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\lib-tk', 'C:\\PROGRA~1\\QGISWI~1\\bin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\PIL', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\jinja2-2.7.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\markupsafe-0.23-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\pytz-2012j-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Pythonwin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Shapely-1.2.18-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\wx-2.8-msw-unicode', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlrd-0.9.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlwt-0.7.5-py2.7.egg', u'C:/Users/Asus A450C/.qgis2//python', 'C:/Users/Asus A450C/.qgis2/python/plugins\\inasafe', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins\\inasafe\\safe_extras\\parameters', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins', 'C:\\Users\\Asus A450C\\.qgis2\\python\\plugins\\inasafe\\safe_extras', 'C:\\PROGRA~1\\QGISWI~1\\apps\\qgis-ltr\\python\\plugins\\fTools\\tools', 'D:/Work/OpenStreetMaps/InaSAFE/InaSAFE Testing 3.4/Testing/Tsunami']```

I using QGIS in Bahasa Indonesia

These are my spesification :

- QGIS 2.8.9 Wien (LTR) in Bahasa Indonesia
- InaSAFE 3.4.0 dev-master
- RAM 8 GB
- Windows 10 64 Bit

**Comment by Gustry** [comment]
This issue should be fixed in develop and.
The master branch will be fixed in a few minutes.

**Comment by harrymahar** [comment]
Thankyou @Gustry , really looking forward to see the new improvement for this problem.

**Comment by harrymahar** [comment]
Hi @Gustry @Charlotte-Morgan @samnawi @felix-yew @ismailsunni , Today I try using InaSAFE 3.4.0 Minimum Needs Manager using Bahasa Indonesia in QGIS 2.14.3 Essen and I still got python error when I want to **Add New Item** and **Edit Item**. It said like this :

```
An error has occurred while executing Python code: 

KeyError: u'Units' 
Traceback (most recent call last):
  File "C:/Users/Asus A450C/.qgis2/python/plugins\inasafe\safe\gui\tools\minimum_needs\needs_manager_dialog.py", line 588, in save_resource
    resource)
  File "C:/Users/Asus A450C/.qgis2/python/plugins\inasafe\safe\gui\tools\minimum_needs\needs_profile.py", line 325, in format_sentence
    resource[replace],
KeyError: u'Units'


Versi Python: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)] 
Versi QGIS: 2.14.3-Essen Essen, cf2ebb8 

Python Path:
C:/PROGRA~1/QGISES~1/apps/qgis/./python/plugins\processing
C:/PROGRA~1/QGISES~1/apps/qgis/./python
C:/Users/Asus A450C/.qgis2/python
C:/Users/Asus A450C/.qgis2/python/plugins
C:/PROGRA~1/QGISES~1/apps/qgis/./python/plugins
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\matplotlib-1.3.1-py2.7-win-amd64.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\nose-1.3.3-py2.7.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\tornado-4.0.1-py2.7-win-amd64.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\backports.ssl_match_hostname-3.4.0.2-py2.7.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\certifi-14.05.14-py2.7.egg
C:\PROGRA~1\QGISES~1\bin\python27.zip
C:\PROGRA~1\QGISES~1\apps\Python27\DLLs
C:\PROGRA~1\QGISES~1\apps\Python27\lib
C:\PROGRA~1\QGISES~1\apps\Python27\lib\plat-win
C:\PROGRA~1\QGISES~1\apps\Python27\lib\lib-tk
C:\PROGRA~1\QGISES~1\bin
C:\PROGRA~1\QGISES~1\apps\Python27
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\PIL
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\jinja2-2.7.2-py2.7.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\markupsafe-0.23-py2.7-win-amd64.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\pytz-2012j-py2.7.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\win32
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\win32\lib
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\Pythonwin
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\Shapely-1.2.18-py2.7-win-amd64.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\wx-2.8-msw-unicode
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\xlrd-0.9.2-py2.7.egg
C:\PROGRA~1\QGISES~1\apps\Python27\lib\site-packages\xlwt-0.7.5-py2.7.egg
C:/Users/Asus A450C/.qgis2//python
C:/Users/Asus A450C/.qgis2/python/plugins\inasafe
C:\Users\Asus A450C\.qgis2\python\plugins\inasafe\safe_extras\parameters
C:\Users\Asus A450C\.qgis2\python\plugins
C:\Users\Asus A450C\.qgis2\python\plugins\inasafe\safe_extras
C:\PROGRA~1\QGISES~1\apps\qgis\python\plugins\fTools\tools
D:/Work/OpenStreetMaps/InaSAFE/InaSAFE Testing 3.4/Testing/EarthQuake
```

Thankyou

**Comment by Gustry** [comment]
@harrymahar Can you try my PR : https://github.com/inasafe/inasafe/pull/2892
Sorry, it doesn't fix the problem for now. I tried to replicate your issue using InaSAFE in French and in Bahasa Indonesia, but for me it's working fine.
Can you copy/paste your readable sentence here to be sure ? and also the error that my PR will give you. Maybe you can also attach your `minimum needs` JSON file available in your .qgis2/minimum_needs folder.
Thanks

**Comment by ismailsunni** [comment]
Trying in develop, I got this error message:
```
Traceback (most recent call last):
  File "/Users/ismailsunni/.qgis2/python/plugins/inasafe-dev/safe/gui/tools/minimum_needs/needs_manager_dialog.py", line 588, in save_resource
    resource)
  File "/Users/ismailsunni/.qgis2/python/plugins/inasafe-dev/safe/gui/tools/minimum_needs/needs_profile.py", line 325, in format_sentence
    resource[replace],
KeyError: u'Units'
```

**Comment by ismailsunni** [comment]
Whoops, I think I know the problem.
It's because, in Bahasa, there is no plural for `unit`, so both `Unit` and `Units` are translated to the same word in Bahasa, `satuan`. Perhaps I will translate `Units` to `satuan (jamak)`

**Comment by ismailsunni** [comment]
Hi @harrymahar 
Please try the fix in #2893 for this issue.

**Comment by harrymahar** [comment]
I have tried Hi @ismailsunni @Gustry , I have tried PR for this issue and It works successfully..

Here I attached screenshoot new minimum needs and run it in InaSAFE.

![minimum needs indo oke](https://cloud.githubusercontent.com/assets/2534179/15771494/65397572-2995-11e6-83c4-f54e8bcf97b3.JPG)
